pkg/cmd/admin/migrate/images/images.go

```diff
@@ -0,0 +1,484 @@
+//   ParseDockerImageReference)
+func ParseMapping(s string) (ImageReferenceMapping, error) {
+	parts := strings.SplitN(s, "=", 2)
+	from := strings.SplitN(parts[0], "/", 2)
+	to := strings.SplitN(parts[1], "/", 2)
+	if len(from) < 2 || len(to) < 2 {
+		return ImageReferenceMapping{}, fmt.Errorf("all arguments must be of the form REGISTRY/NAME=REGISTRY/NAME, where registry or name may be '*' or a value")
+	}
+	if len(from[0]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid source: registry must be specified (may be '*')", parts[0])
+	}
+	if len(from[1]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid source: name must be specified (may be '*')", parts[0])
+	}
+	if len(to[0]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: registry must be specified (may be '*')", parts[1])
+	}
+	if len(to[1]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: name must be specified (may be '*')", parts[1])
+	}
+	if from[0] == "*" {
+		from[0] = ""
+	}
+	if from[1] == "*" {
+		from[1] = ""
+	}
+	if to[0] == "*" {
+		to[0] = ""
+	}
+	if to[1] == "*" {
+		to[1] = ""
+	}
+	if to[0] == "" && to[1] == "" {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: at least one change must be specified", parts[1])
+	}
+	if from[0] == from[1] && to[0] == to[1] {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not valid: must target at least one field to change", s)
+	}
+	return ImageReferenceMapping{
+		FromRegistry: from[0],
+		FromName:     from[1],
+		ToRegistry:   to[0],
+		ToName:       to[1],
+	}, nil
+}
+
+// ImageReferenceMappings provide a convenience method for transforming an input reference
+type ImageReferenceMappings []ImageReferenceMapping
+
+// MapReference transforms the provided Docker image reference if any mapping matches the
+// input. If the reference cannot be parsed, it will not be modified.
+func (m ImageReferenceMappings) MapReference(in string) string {
+	ref, err := imageapi.ParseDockerImageReference(in)
+	if err != nil {
+		return in
+	}
+	registry := ref.Registry
+	if len(registry) == 0 {
+		registry = "docker.io"
+	}
+	name := ref.RepositoryName()
+	for _, mapping := range m {
+		if len(mapping.FromRegistry) > 0 && mapping.FromRegistry != registry {
+			continue
+		}
+		if len(mapping.FromName) > 0 && mapping.FromName != name {
+			continue
+		}
+		if len(mapping.ToRegistry) > 0 {
+			ref.Registry = mapping.ToRegistry
+		}
+		if len(mapping.ToName) > 0 {
+			ref.Namespace = ""
+			ref.Name = mapping.ToName
+		}
+		return ref.Exact()
+	}
+	return in
+}
+
+// MapDockerAuthKey transforms the provided Docker Config host key if any mapping matches
+// the input. If the reference cannot be parsed, it will not be modified.
+func (m ImageReferenceMappings) MapDockerAuthKey(in string) string {
+	value := in
+	if len(value) == 0 {
+		value = "https://index.docker.io/v1/"
```

Comment:
It's not actually the same, `/v1/` is a special construct that is required for auth stuff.